File: app/src/main/java/me/shreyasr/chatse/chat/ChatRoom.java
package me.shreyasr.chatse.chat;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Represents a room: a site and a room number. Immutable.
 */
public class ChatRoom implements Parcelable {

    public final String site;
    public final int num;

    public ChatRoom(String site, int num) {
        this.site = site;
        this.num = num;
    }

    protected ChatRoom(Parcel in) {
        site = in.readString();
        num = in.readInt();
    }

    @Override public String toString() {
        return "Room " + num;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(site);
        dest.writeInt(num);
    }

    @SuppressWarnings("unused")
    public static final Parcelable.Creator<ChatRoom> CREATOR = new Parcelable.Creator<ChatRoom>() {
        @Override
        public ChatRoom createFromParcel(Parcel in) {
            return new ChatRoom(in);
        }

        @Override
        public ChatRoom[] newArray(int size) {
            return new ChatRoom[size];
        }
    };
}

File: app/src/main/java/me/shreyasr/chatse/chat/ChatRoom.kt
package me.shreyasr.chatse.chat

import android.os.Parcel
import android.os.Parcelable

/**
 * Represents a room: a site and a room number. Immutable.
 */
class ChatRoom : Parcelable {
    val site: String
    val num: Int

    constructor(site: String, num: Int) {
        this.site = site
        this.num = num
    }

    protected constructor(`in`: Parcel) {
        site = `in`.readString()
        num = `in`.readInt()
    }

    override fun toString(): String {
        return "Room " + num
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(site)
        dest.writeInt(num)
    }

    companion object {

        val CREATOR: Parcelable.Creator<ChatRoom> = object : Parcelable.Creator<ChatRoom> {
            override fun createFromParcel(`in`: Parcel): ChatRoom {
                return ChatRoom(`in`)
            }

            override fun newArray(size: Int): Array<ChatRoom?> {
                return arrayOfNulls(size)
            }
        }
    }
}